pandas/core/sparse/frame.py

```diff
@@ -191,11 +191,13 @@ def _init_spmatrix(self, data, index, columns, dtype=None,
         for col, rowvals in values.groupby(data.col):
             # get_blocks expects int32 row indices in sorted order
             rows = rowvals.index.values.astype(np.int32)
+            vals = np.array([y for x, y in sorted(rowvals.iteritems())],
```

Comment:
@jreback 
Is this solution still problematic? I think sorting the values by their index ensures that the values are in the correct order that aligns them with the index being passed below.